osgar/node.py

```diff
@@ -31,7 +31,11 @@ def is_bus_alive(self):
     def update(self):
         timestamp, channel, data = self.bus.listen()
         self.time = timestamp
-        setattr(self, channel, data)
+        handler = getattr(self, "on_" + channel, None)
+        if handler is not None:
+            handler(data)
+        else:
+            assert False, channel  # unknown
```

Comment:
This code will break a lot of things (not only here). We should consider put the "setattr" part to "else" section. Maybe with some warning message like "this feature will no longer supported..". Transitional period.